.github/workflows/docs_deploy.yml

```diff
@@ -0,0 +1,224 @@
+          python-version: '3.9'
+
+      - name: Install dependencies
+        run: tools/install_ubuntu_docs_dependencies.sh
+
+      # Sync with '.azure/tutorials-linux.yml'.
+      - name: Download current tutorials
+        run: tools/prepare_tutorials.bash algorithms circuits circuits_advanced operators
+        shell: bash
+
+      # This is just to have tox create the environment, so we can use it to execute the tutorials.
+      # We want to re-use it later for the build, hence 'tox run --notest' instead of 'tox devenv'.
+      - name: Prepare Python environment
+        run: tox run -e docs --notest
+
+      # The reason to use the custom script rather than letting 'nbsphinx' do its thing normally
+      # within the Sphinx build is so that the execution process is the same as in the test CI.
+      - name: Execute tutorials in place
+        run: .tox/docs/bin/python tools/execute_tutorials.py docs/tutorials
+        env:
+          QISKIT_CELL_TIMEOUT: "300"
+
+      - name: Build documentation
+        # We can skip re-installing the package, since we just did it a couple of steps ago.
+        run: tox run -e docs --skip-pkg-install
+        env:
+          QISKIT_ENABLE_ANALYTICS: "true"
+          # We've already built them.
+          QISKIT_DOCS_BUILD_TUTORIALS: "never"
+
+      - name: Build translatable strings
+        run: tox -e gettext
+        env:
+          # We've already built them.
+          QISKIT_DOCS_BUILD_TUTORIALS: "never"
+
+      - name: Store built documentation artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: qiskit-docs
+          path: |
+            ./docs/_build/html/*
+            !**/.doctrees
+            !**/.buildinfo
+          if-no-files-found: error
+
+      - name: Store translatable strings artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: qiskit-translatables
+          path: ./docs/locale/en/*
+          if-no-files-found: error
+
+  deploy:
+    if: github.event_name != 'workflow_dispatch' || inputs.do_deployment
+    name: Deploy to qiskit.org
+    needs: [build]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          path: qiskit
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: qiskit-docs
+          path: deploy
+
+      - id: choose
+        name: Choose deployment location
+        run: |
+          set -e
+          case ${{ github.event_name }} in
+            push)
+              case ${{ github.ref_name }} in
+                main)
+                  echo "deploy_prefix=dev" >> "$GITHUB_OUTPUT"
+                  ;;
+                # This currently deploys to the documentation root on a push to _any_ stable branch;
+                # when we're supporting more than one version at once, it will need revisiting.
+                stable/*)
+                  echo "deploy_prefix=" >> "$GITHUB_OUTPUT"
+                  ;;
```

Comment:
So any push to `stable/*` will update the main published version of the docs? I'm a bit concerned that for somethings we might be changing the docs and undoing it for things like release notes as they'll be auto-generated via reno in the near future. Also for release candidates we're working on stable before we've released the version so the docs on the main page will reflect this. I'm thinking it might be better to reserve this for only on tag events and workflow dispatch.

Reply:
Oh yeah, I didn't think about the release-candidate phase.

I'll have to rewrite the logic a bit here to allow more than one deployment to happen from one event; we'll need `tag` to trigger two different uploads with your strategy (which is fine, I just need to rewrite the logic).  Previously the metapackage had similar-ish logic to what I've written, in that any push to `master` triggered a deployment to `/`, so there wasn't actually ever a double deployment from one action trigger, it's just that a push and a tag both triggered deployments.

Reply:
Done in fc27c4a8f.